lighthouse-core/audits/legacy-javascript.js

```diff
@@ -161,88 +166,98 @@ class LegacyJavascript extends Audit {
+      {...defineModules('es6.reflect.get'), name: 'Reflect.get'},
+      {...defineModules('es6.reflect.get-own-property-descriptor'), name: 'Reflect.getOwnPropertyDescriptor'},
+      {...defineModules('es6.reflect.get-prototype-of'), name: 'Reflect.getPrototypeOf'},
+      {...defineModules('es6.reflect.has'), name: 'Reflect.has'},
+      {...defineModules('es6.reflect.is-extensible'), name: 'Reflect.isExtensible'},
+      {...defineModules('es6.reflect.own-keys'), name: 'Reflect.ownKeys'},
+      {...defineModules('es6.reflect.prevent-extensions'), name: 'Reflect.preventExtensions'},
+      {...defineModules('es6.reflect.set'), name: 'Reflect.set'},
+      {...defineModules('es6.reflect.set-prototype-of'), name: 'Reflect.setPrototypeOf'},
+      {...defineModules('es6.set'), name: 'Set'},
+      {...defineModules('es6.string.code-point-at'), name: 'String.prototype.codePointAt'},
+      {...defineModules('es6.string.ends-with'), name: 'String.prototype.endsWith'},
+      {...defineModules('es6.string.from-code-point'), name: 'String.fromCodePoint'},
+      {...defineModules('es6.string.includes'), name: 'String.prototype.includes'},
+      {...defineModules('es6.string.raw'), name: 'String.raw'},
+      {...defineModules('es6.string.repeat'), name: 'String.prototype.repeat'},
+      {...defineModules('es6.string.starts-with'), name: 'String.prototype.startsWith'},
+      {...defineModules('es6.string.trim'), name: 'String.prototype.trim'},
+      {coreJs2Module: 'es6.typed.array-buffer', coreJs3Module: 'es.array-buffer.constructor', name: 'ArrayBuffer'},
+      {coreJs2Module: 'es6.typed.data-view', coreJs3Module: 'es.data-view', name: 'DataView'},
+      {...defineModules('es6.typed.float32-array'), name: 'Float32Array'},
+      {...defineModules('es6.typed.float64-array'), name: 'Float64Array'},
+      {...defineModules('es6.typed.int16-array'), name: 'Int16Array'},
+      {...defineModules('es6.typed.int32-array'), name: 'Int32Array'},
+      {...defineModules('es6.typed.int8-array'), name: 'Int8Array'},
+      {...defineModules('es6.typed.uint16-array'), name: 'Uint16Array'},
+      {...defineModules('es6.typed.uint32-array'), name: 'Uint32Array'},
+      {...defineModules('es6.typed.uint8-array'), name: 'Uint8Array'},
+      {...defineModules('es6.typed.uint8-clamped-array'), name: 'Uint8ClampedArray'},
+      {...defineModules('es6.weak-map'), name: 'WeakMap'},
+      {...defineModules('es6.weak-set'), name: 'WeakSet'},
+      {...defineModules('es7.array.includes'), name: 'Array.prototype.includes'},
+      {...defineModules('es7.object.entries'), name: 'Object.entries'},
+      {...defineModules('es7.object.get-own-property-descriptors'), name: 'Object.getOwnPropertyDescriptors'},
+      {...defineModules('es7.object.values'), name: 'Object.values'},
+      {...defineModules('es7.string.pad-end'), name: 'String.prototype.padEnd'},
+      {...defineModules('es7.string.pad-start'), name: 'String.prototype.padStart'},
```

Comment:
drive by: a simpler array and then a `.map()` into a array of `{name: string, coreJs2Module: string, coreJs3Module: string}` might read more straightforwardly here.

Something like
```js
static getPolyfillData() {
  return [
    /* eslint-disable max-len */
    ['Array.prototype.fill', 'es6.array.fill'],
    ['Array.prototype.filter', 'es6.array.filter'],
    // ...
    ['String.prototype.trim', 'es6.string.trim'],
    // These break the coreJs2/coreJs3 naming pattern so are set explicitly.
    {name: 'ArrayBuffer', coreJs2Module: 'es6.typed.array-buffer', coreJs3Module: 'es.array-buffer.constructor'},
    {name: 'DataView', coreJs2Module: 'es6.typed.data-view', coreJs3Module: 'es.data-view'},
    ['Float32Array', 'es6.typed.float32-array'],
    // ...
    ['String.prototype.padStart', 'es7.string.pad-start'],
    /* eslint-enable max-len */
  ].map(info => {
    if (!Array.isArray(info)) return info;

    const [name, coreJs2Module] = info;
    return {
      name,
      coreJs2Module,
      coreJs3Module: coreJs2Module
        .replace('es6.', 'es.')
        .replace('es7.', 'es.')
        .replace('typed.', 'typed-array.'),
    };
  });
}
```
(could also do an `in` check or whatever instead of `isArray` if want to keep named properties)

Reply:
nice :)